core/trino-spi/src/main/java/io/trino/spi/Page.java

```diff
@@ -43,6 +44,11 @@ static Page wrapBlocksWithoutCopy(int positionCount, Block[] blocks)
         return new Page(false, positionCount, blocks);
     }
 
+    public static long getInstanceSizeInBytes(int blockCount)
```

Comment:
nit: move above `wrapBlocksWithoutCopy`.

Use it from `io.trino.spi.Page#updateRetainedSize`?